cli/commands/run/index.ts

```diff
@@ -11,16 +12,26 @@ import { RunProdServer } from './server';
 
 export default function provideRun(
   container: AwilixContainer,
+  ethersProvider: providers.JsonRpcProvider,
+  chainIds: number[],
   cache: Cache,
   args: any
 ): CommandHandler {
   assertExists(container, 'container')
   assertExists(cache, 'cache')
+  assertExists(chainIds, 'chainIds')
+  assertExists(ethersProvider, "ethersProvider");
   assertExists(args, 'args')
 
   return async function run(runtimeArgs: any = {}) {
     args = { ...args, ...runtimeArgs }
 
+    const network = await ethersProvider.getNetwork();
+
+    if(!network) throw Error(`No network detected at rpc url.`);
+
+    if(!chainIds.includes(network.chainId)) throw Error(`Detected chainId [${network.chainId}] is missing from the package.json.`)
```

Comment:
I would also mention jsonRpcUrl here so that devs will know where the issue is coming from i.e. `Detected chainId mismatch between jsonRpcUrl (chainId: 1) and package.json (chainIds: 2, 5, 7)`